tests/support/pretest.js

```diff
@@ -0,0 +1,142 @@
+	'testharnessreport.js'
+];
+
+var reqOpts = {
+	'headers': {
+		'Authorization': 'token ' + atob(token),
+		'User-Agent': 'request'
+	}
+};
+
+var options = {
+	'raw': function(entry) {
+		return defaultsDeep({
+			'url': entry,
+			'headers': { 'Accept': 'application/vnd.github.v3.raw' }
+		}, reqOpts);
+	},
+	'tree': function(entry) {
+		if (url.parse(entry).hostname !== 'api.github.com') {
+			entry = _.template(treeUrl)(parseRepoString(entry));
+		}
+		return defaultsDeep({ 'url': entry, 'json': true }, reqOpts);
+	}
+};
+
+var defaultsDeep = _.partialRight(_.merge, function defaults(objVal, srcVal) {
+	return objVal === undefined ? srcVal : _.merge(objVal, srcVal, defaults);
+});
+
+function parseRepoString(string) {
+	var parts = String(string).match(/([^\/]+)\/([^#]+)#([\s\S]+)/);
+	return {
+		'owner': _.result(parts, 1, ''),
+		'repo': _.result(parts, 2, ''),
+		'sha': _.result(parts, 3, '')
+	};
+}
+
+function getRaw(entry) {
+	return request.getAsync(options.raw(entry)).then(
+		function(res) { return _.result(res, 1); },
+		function(err) { throw err; }
+	);
+}
+
+function getTree(entry) {
+	return request.getAsync(options.tree(entry)).then(
+		function(res) { return _.result(res, '[1].tree'); },
+		function(err) { throw err; }
+	);
+}
+
+function getFiles(tree) {
+	var pe = _.find(tree, 'path', 'pointerevents'),
+		resources = _.find(tree, 'path', 'resources');
+
+	return Promise.all([
+		getTree(pe.url)
+			.then(getTests),
+		getTree('w3c/testharness.js#' + resources.sha)
+			.then(getHarness)
+	]);
+}
+
+function getTests(tree) {
+	return Promise.all(_.map(tree, function(object) {
+		var $raw = getRaw(object.url);
+		return $raw.then(function(raw) {
+			return fs.outputFileAsync(path.join(testPath, object.path), raw, 'utf-8');
+		});
+	}));
+}
+
+function getHarness(tree) {
+	return Promise.all(_.map(harnessFiles, function(hf) {
+		var object = _.find(tree, 'path', hf),
+			$raw = getRaw(object.url);
+
+		return $raw.then(function(raw) {
+			return fs.outputFileAsync(path.join(testPath, 'resources', object.path), raw, 'utf-8');
+		});
+	}));
+}
+
+function modFiles() {
+	return glob(path.join(testPath, '*.html')).then(function(paths) {
+		return Promise.all(_.map(paths, function(path) {
+			return fs.readFileAsync(path, 'utf8')
+				.then(modFile)
+				.then(function(source) {
+					return fs.writeFileAsync(path, source, 'utf-8');
+				});
+		}));
+	});
+}
+
+function modFile(source) {
+
+	// Ensure pep.js is the first script loaded on the page
+	source = source.replace(/^[\t ]*(?=<script\b)/im, '\n$&<script src="' + encodeURI(pepPath) + '"></script>\n$&');
+
+	// Make paths to scripts and style sheets relative instead of absolute
+	return source.replace(/((?:src|href)\s*=\s*['"])([^.])/g, function(match, prelude, chr) {
+		return prelude + '.' + (chr === '/' ? '' : '/') + chr;
+	});
+}
+
+module.exports = function() {
+	var parsed = parseRepoString(testPkg.repo);
+	if (fs.existsSync(testPath) && fs.existsSync(shaPath) &&
```

Comment:
> On a side note, fs.exists() is discouraged since it hides errors, but it's fine for this case since an error will just result in fetching the files.

Should we move away from fs.exists? It's [deprecated](https://github.com/nodejs/io.js/issues/103).